File: SkyEditor.SaveEditor/PluginDefinition.vb
Imports SkyEditor.Core

Public Class PluginDefinition
    Inherits SkyEditor.Core.SkyEditorPlugin

    Public Overrides ReadOnly Property Credits As String
        Get
            Return My.Resources.Language.PluginCredits
        End Get
    End Property

    Public Overrides ReadOnly Property PluginAuthor As String
        Get
            Return My.Resources.Language.PluginAuthor
        End Get
    End Property

    Public Overrides ReadOnly Property PluginName As String
        Get
            Return My.Resources.Language.PluginName
        End Get
    End Property

    Public Overrides Sub Load(manager As PluginManager)
        MyBase.Load(manager)
        manager.RegisterIOFilter("*.sav", My.Resources.Language.RawSaveFile)
        manager.RegisterIOFilter("*.skypkm", My.Resources.Language.SkyPkmFile)
        manager.RegisterIOFilter("*.skypkmex", My.Resources.Language.SkyPkmExFile)
        manager.RegisterIOFilter("*.skypkmq", My.Resources.Language.SkyPkmQFile)
        manager.RegisterIOFilter("*.tdpkm", My.Resources.Language.TDPkmFile)
        manager.RegisterIOFilter("*.tdpkmex", My.Resources.Language.TDPkmExFile)
        manager.RegisterIOFilter("*.rbpkm", My.Resources.Language.RBPkmFile)
        manager.RegisterIOFilter("game_data", My.Resources.Language.GameDataFile)
    End Sub
End Class

File: SkyEditor.SaveEditor/PluginDefinition.cs
using SkyEditor.Core;
using SkyEditor.SaveEditor.Resources;
using System;
using System.Collections.Generic;
using System.Text;

namespace SkyEditor.SaveEditor
{
    public class PluginDefinition : SkyEditorPlugin
    {
        public override string PluginName => Language.PluginName;

        public override string PluginAuthor => Language.PluginAuthor;

        public override string Credits => Language.PluginCredits;

        public override void Load(PluginManager manager)
        {
            base.Load(manager);

            manager.RegisterIOFilter("*.sav", Language.RawSaveFile);
            manager.RegisterIOFilter("*.skypkm", Language.SkyPkmFile);
            manager.RegisterIOFilter("*.skypkmex", Language.SkyPkmExFile);
            manager.RegisterIOFilter("*.skypkmq", Language.SkyPkmQFile);
            manager.RegisterIOFilter("*.tdpkm", Language.TDPkmFile);
            manager.RegisterIOFilter("*.tdpkmex", Language.TDPkmExFile);
            manager.RegisterIOFilter("*.rbpkm", Language.RBPkmFile);
            manager.RegisterIOFilter("game_data", Language.GameDataFile);
        }
    }
}
